php-frontend/src/main/java/org/sonar/plugins/php/api/symbols/QualifiedName.java

```diff
@@ -59,16 +60,16 @@ public static QualifiedName create(String... names) {
     int length = names.length;
     if (length == 0) {
       throw new IllegalStateException("Cannot create an empty qualified name");
-    } else if (length == 1) {
-      return new QualifiedName(Collections.emptyList(), names[0]);
     } else {
-      return new QualifiedName(Arrays.asList(names));
+      return new QualifiedName(ImmutableList.copyOf(names));
     }
   }
 
   public static QualifiedName create(NamespaceNameTree nameTree) {
-    List<String> namespaces = nameTree.namespaces().stream().map(NameIdentifierTree::text).collect(Collectors.toList());
-    return new QualifiedName(namespaces, nameTree.name().text());
+    ImmutableList.Builder<String> newName = ImmutableList.builder();
+    nameTree.namespaces().stream().map(NameIdentifierTree::text).forEach(newName::add);
+    newName.add(nameTree.name().text());
+    return new QualifiedName(newName.build());
```

Comment:
This change seems redundant with the other ones in the file: the constructors now handle the lowercase conversion.